src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -71,11 +78,86 @@ You can use headers like dynamic API keys in your function, which give you acces
 Learn more about dynamic API keys
 {% /arrow_link %}
 
+# Hot reload {% #hot-reload %}
+
+By default, Appwrite CLI hot-reloads your functions, which means you can update the function code and see the changes happen in real time without restarting. Compiled languages must translate the source code into machine code, causing the system to rebuild continuously. However, interpreted languages only refreshes the function when you change the dependency file. 
```

Comment:
> interpreted languages only refreshes the function when you change the dependency file. 

This makes it sound like the function will only update if the dependency file changes, but that's not the case. I also misunderstood what actually happens and where the dependency file comes into play. There are actually 3 cases/scenarios:

1. compiled language + any change
2. interpreted language + change to dependency file
3. interpreted language + change to code

So, this section should look something like:

```suggestion
By default, the Appwrite CLI hot-reloads your functions, which means you can update the function code and the changes will be applied automatically. How this happens differs between runtimes with compiled languages versus interpreted ones.

Because runtimes with compiled languages must translate the source code into machine code, the function must rebuild on change.

When the source code in a runtime with an interpreted languages is updated, the function only needs to restart with the updated file. However, if a dependency file for the interpreted language is updated, the function must rebuild to update the dependencies. Refer to the table below for the dependency files of each language.
```